core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/part/EnvelopePart.java

```diff
@@ -575,6 +575,9 @@ public EnvelopePart slice(
             int beginStepIndex, double beginPosition, double beginSpeed,
             int endStepIndex, double endPosition, double endSpeed
     ) {
+        if (stepCount() == 0)
+            return null;
+
```

Comment:
This should only happen if the sliced envelope part has a single point (one position / one speed / no timedelta). It shouldn't return null but the envelope part itself, assuming the begin/end position/speed match (we could add asserts for that).